contracts/Splitter.sol

```diff
@@ -7,34 +7,29 @@ contract Splitter {
     
     address payable[] internal recipients = [0x20601F1Ddb44E28b1511EdBD316A368D80116408, //"Bob";
                                             0x24F1500890505ceD8534502ab403F20ce99feea0]; //"Carol";
+    uint InitialValue = 100 ether;
 
     event Deposit(uint date,
                   address indexed _from,
                   address indexed _to,
                   uint amount);
 
     function deposit() external payable{
-        doDeposit(msg.sender, msg.value);
-    }
 
-    function doDeposit(address _sender, uint _value) public payable {
-        if(_sender != sender){
+        if(msg.value % 2 != 0) {
+            revert("Only even numbers can be sent");
+        }
+
+        if(msg.sender != sender){
             revert("Only Alice can send ether");
         }
-        
-        uint split = split(_value);
-        
+
+        uint split = msg.value / 2;
+
         for (uint i=0; i < recipients.length; i++) {
             recipients[i].transfer(split);
 
-            emit Deposit(now, _sender, recipients[i], split);
-        }
-    }
-
-    function split(uint amount) internal pure returns(uint) {
-        if(amount % 2 != 0) {
-            revert("This odd number cannot be split");
+            emit Deposit(now, msg.sender, recipients[i], split);
```

Comment:
I see, so two events are emitted. 

If you knock it down so two addresses are passed into this function then you will flatten it out, something like

```
emit SplitFunds(msg.sender, msg.value, receiver1, receiver2);
```